### ISSUE TITLE
Inference time taking too long - Speaker diarization 

### ISSUE THREAD

**Issue by pedrohenriqp** [pyannote/pyannote-audio#876]
Hi guys, I have a question about inference time using pipeline for Speaker Diarization. I I trained a model with my own data, and I am using pipeline to make predictions in a new .wav file. A 12 minute long conversation audio is taking about 14 minutes for inference.

Is that right? If yes, do you guys have any tips to speed up the pipeline inference time?

I am using this kind of code snippet to run inference. Is there a better way to run  inferences on many files without needing to use _for_ loops?

```python
demofile_path = "audio/file_example.wav"
DEMO_FILE = {'uri': 'file_example', 'audio': demofile_path}
pipeline = SpeakerDiarization(sad_scores = "models/speech_activity_detection/train/SpeakerDiarization.MixHeadset.train/weights/0022.pt",
                              scd_scores = "models/speaker_change_detection/train/SpeakerDiarization.MixHeadset.train/weights/0001.pt",
                              embedding = "models/speaker_embedding/train/SpeakerDiarization.MixHeadset.train/weights/0010.pt", 
                              method= "affinity_propagation")

pipeline.load_params("pipelines/speaker_diarization/train/SpeakerDiarization.MixHeadset.development/params.yml")

diarization = pipeline(DEMO_FILE)
```
```bash
pip freeze | grep pyannote

pyannote.audio==1.1.1
pyannote.core==4.3      
pyannote.database==4.1.1
pyannote.metrics==3.2   
pyannote.pipeline==1.5.2
```

Thanks

**Comment by Yagna24** [comment]
Yes, on colab gpu for just running `diarization = pipeline(audio.wav)` it takes more than 9 minutes. 
Please let me know if u come across the solution.

**Comment by WilledgeR** [comment]
Because this repo is written by a group of **noobies programmers** (just elementary AI students).
There is tons of **useless polymorphism implementations**, complicated class inheritance that makes debugging so hard.

Any way I spend a week and Refactored this repo and made a **"ahead of real time"** version that runs fully on gpu.

I actually wonder how they call the module "Pipeline" while it's neither **modular** or **realtime**.
most of the code is written in numpy, and is a good case of how to use speech brain in your project as an example and nothing more.

**Comment by lutharsanen** [comment]
@WilledgeR, can your "ahead of real time" version be found somewhere? I run also into trouble, that the speaker diarization does not use the available gpu ressources.

best, 

Lutharsanen
